File: 0027. Remove Element/solution_2.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */

var removeElement = function(nums, val) 
{
    const newNums = new Array
    let find = 0;
    nums.forEach(n => {
        if (n != val) newNums.push(n)
        else find++;
    });

    nums.splice(0, newNums.length, ...newNums)

    return nums.length - find
};

File: 0027. Remove Element/solution_2.cs
public class Solution
{
    public int RemoveElement(int[] nums, int val)
    {
        List<int> newNums = new List<int>();
        int find = 0;

        foreach (int n in nums)
            if (n != val) newNums.Add(n);
            else find++;

        for (int i = 0; i < newNums.Count; ++i)
            nums[i] = newNums[i];

        return nums.Length - find;
    }
}